vm/systemSmartContracts/delegation.go

```diff
@@ -1285,13 +1349,16 @@ func (d *delegation) claimRewards(args *vmcommon.ContractCallInput) vmcommon.Ret
 		return vmcommon.UserError
 	}
 
-	err = d.eei.Transfer(args.CallerAddr, args.RecipientAddr, delegator.UnClaimedRewards, nil, 0)
+	//TODO: The call of method saveDelegatorData should be done before the call of method Transfer, because if it is done afterwards and it will fail,
+	//then the transfer of UnClaimedRewards remains done but the update of (delegator.UnClaimedRewards and delegator.RewardsCheckpoint)
+	//will not be done. This means that if the claimRewards method would be called again, it will transfer these unClaimedRewards one more time?
```

Comment:
nothing will remain. Any error invalidates all previous changes. save is done in a cache - cache is invalidated if returnCode != vmcommon.Ok

Reply:
ok